keptn/keptn
# feat(installer): Add option to define nodeSelector globally or per service

## This PR

Adds the possibility to set nodeSelectors for all keptn pods

### Related Issues

Fixes #2887

### Notes

Although the DoD in issue #2887 states that the default nodeSelector `kubernetes.io/os: linux` should be set, i rather would prefer to set no default, because it seems to be best practice in most of the charts (for example also nats and mongodb)

After this PR you could set nodeSelectors quite easily. For example to set `kubernetes.io/os: linux` for control-plane use:

```yaml
control-plane:
  common:
    nodeSelector:
      kubernetes.io/os: linux
  nats:
    nodeSelector:
      kubernetes.io/os: linux
  mongo:
    nodeSelector:
      kubernetes.io/os: linux
```

Additionally you could also overwrite nodeSelectors per service.

### How to test

Without setting any values for nodeSelectors nothing should change (the output of `helm template .` should be the same before and after merging this PR). To see nodeSelectors use `helm template --set control-plane.common.nodeSelector.testkey=testvalue`
